### PR TITLE
Master

### PR DESCRIPTION
**Pull request by randomdavis**
Adds job opening request/approval/denial/fetching functionality

### CODE REVIEW

**Comment by randomdavis** [comment]
I know my tests have a lot of redundant code. Ignore that for now if you please.

File: AES.OpeningsSvc/OpeningSvc.svc.cs [hunk 1]
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
...
+        {
+            return GetOpenings(StoreID, OpeningStatus.PENDING_APPROVAL);
+        }
+
+        public List<JobOpeningContract> GetRejectedOpenings(int StoreID)
+        {
+            return GetOpenings(StoreID, OpeningStatus.REJECTED);
+        }
+
+        public bool RequestOpening(int StoreID, JobOpeningContract opening)
```
Comment (DavidA94) [inline AES.OpeningsSvc/OpeningSvc.svc.cs]: How will n openings be requested? Will they have to call this n times?

File: AES.OpeningsSvc/OpeningSvc.svc.cs [hunk 2]
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
...
+                var openingRequest = new JobOpening()
+                {
+                    Store = store,
+                    Job = job,
+                    Status = OpeningStatus.PENDING_APPROVAL,
+                    RequestNotes = opening.RequestNotes
+                };
+
+                db.JobOpenings.Add(openingRequest);
+                db.SaveChanges();
```
Comment (DavidA94) [inline AES.OpeningsSvc/OpeningSvc.svc.cs]: Good idea to ensure that this saves by ensuring the number returned is >0.

Comment (randomdavis) [inline AES.OpeningsSvc/OpeningSvc.svc.cs]: Fixed in my next push

File: AES.OpeningsSvc/OpeningSvc.svc.cs (hunk 1 above)
Comment (randomdavis) [inline AES.OpeningsSvc/OpeningSvc.svc.cs]: Fixed in my next push

File: AES.OpeningsSvc/OpeningSvc.svc.cs [hunk 3]
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
...
+                {
+                    if (db.SaveChanges() < 1)
+                    {
+                        return false;
+                    }
+                }
+                catch (System.Exception ex)
+                {
+
+                    throw ex;
```
Comment (DavidA94) [inline AES.OpeningsSvc/OpeningSvc.svc.cs]: Theoretically, the service should never throw an error. Plus, the catch -> throw makes this try/catch pointless. Either return false or remove the try.

File: AES.OpeningsSvc/OpeningSvc.svc.cs [hunk 4]
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
...
-
                 foreach (var opening in gottenOpenings)
                 {
                     returnedList.Add(new JobOpeningContract(opening));
                 }
             }
+            return returnedList;
+        }
 
+        private List<JobOpeningContract> GetOpenings(int StoreID, OpeningStatus status)
```
Comment (DavidA94) [inline AES.OpeningsSvc/OpeningSvc.svc.cs:123]: Good use of helper method.

File: Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs [hunk 5]
```diff
@@ -126,5 +127,231 @@ public void OpeningSvc_GetOpenings()
...
+
+                db.SaveChanges();
+
+                db.Jobs.RemoveRange(db.Jobs.Where(j => j.JobID == newJob.JobID));
+
+                db.SaveChanges();
+
+                db.Stores.RemoveRange(db.Stores.Where(st => st.ID == newStore.ID));
+
+                db.SaveChanges();
```
Comment (DavidA94) [inline Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs:191]: No point to hit the DB 3 times. RemoveAll -> Save.
Ditto on the other ones.

File: Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs [hunk 6]
```diff
@@ -126,5 +127,231 @@ public void OpeningSvc_GetOpenings()
...
```
Comment (DavidA94) [inline Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs]: Ditto.

File: Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs [hunk 7]
```diff
@@ -126,5 +127,231 @@ public void OpeningSvc_GetOpenings()
...
```
Comment (DavidA94) [inline Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs]: Ditto.

File: AES.OpeningsSvc/OpeningSvc.svc.cs (hunk 3 above)
Comment (randomdavis) [inline AES.OpeningsSvc/OpeningSvc.svc.cs]: Removed t/c in next push

File: AES.OpeningsSvc/OpeningSvc.svc.cs (hunk 4 above)
Comment (randomdavis) [inline AES.OpeningsSvc/OpeningSvc.svc.cs:123]: Thanks!

File: Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs (hunk 5 above)
Comment (randomdavis) [inline Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs:191]: I don't want to remove all openings, just the ones that match the filter I gave. Still, I removed the redundant saves in my next push.

**Pull request merged**